### PR TITLE
Auth config changes

### PR DESCRIPTION
**Pull request by databyjp**
### Why:

Remove refs to 'config.yaml' or similar which are deprecated

### What's being changed:

Removed references to configuration yaml or json files in Configuration/Authentication and Configuration/Authorization pages; replaced them with references to environment variables in the Docker Compose YAML. 

### Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature or enhancements (non-breaking change which adds functionality)
- [x] Documentation updates (non-breaking change which updates documents)

### CODE REVIEW

File: developers/weaviate/current/configuration/authentication.md [hunk 1]
```diff
@@ -39,16 +39,15 @@ any request without an allowed authentication scheme will return `401
 Unauthorized`.
 
 ### Configuration
-Anonymous Access can be configured like so in the respective environment in the
-`configuration.yaml`:
-
-*Note that this example uses yaml, however configuration can also be written as
-json.*
+Anonymous Access can be configured like so in the Docker Compose yaml file like so:
```
Comment (dirkkul) [inline developers/weaviate/current/configuration/authentication.md:42]: double "like so"

File: developers/weaviate/current/configuration/authentication.md [hunk 2]
```diff
@@ -107,33 +105,33 @@ authentication:
...
+
+      # username_claim (required) tells weaviate which claim to use for extracting
+      # the username. The username will be passed to the authorization module.      
+      AUTHENTICATION_OIDC_USERNAME_CLAIM: 'email'
+
+      # groups_claim (optional) tells weaviate which claim to use for extracting
+      # the groups. Groups must be an array of string. If groups_claim is not set
+      # weaviate will not try to extract groups and pass an empty array to the 
+      # authorization module.      
+      AUTHENTICATION_OIDC_GROUPS_CLAIM: 'groups'
```
Comment (dirkkul) [inline developers/weaviate/current/configuration/authentication.md:128]: afaik the groups do not really work. So maybe we should remove this for now? @byronvoorbach

**Review by dirkkul — APPROVED** [review]

File: developers/weaviate/current/configuration/authentication.md [hunk 3]
```diff
@@ -85,19 +84,18 @@ the subject mentioned in the token.
...
+services:
+  weaviate:
+    ...
+    environment:
+      ...
       # enabled (optional - defaults to false) turns OIDC auth on. All other fields in
-      # this section will only be validated if enabled is set to true.
-      enabled: true
+      # this section will only be validated if enabled is set to true.      
+      AUTHENTICATION_OIDC_ENABLED: 'true'
```
Comment (dirkkul) [inline developers/weaviate/current/configuration/authentication.md:98]: better double check if the names of the variables should be changed

File: developers/weaviate/current/configuration/authentication.md (hunk 1 above)
Comment (databyjp) [reply developers/weaviate/current/configuration/authentication.md:42]: Thank you, have fixed it.

File: developers/weaviate/current/configuration/authentication.md (hunk 3 above)
Comment (databyjp) [reply developers/weaviate/current/configuration/authentication.md:98]: Yup. I grabbed them from https://weaviate.io/developers/weaviate/current/installation/docker-compose.html#environment-variables. Is that what you mean?

Comment (dirkkul) [reply developers/weaviate/current/configuration/authentication.md:98]: yes exactly :) Thanks!

File: developers/weaviate/current/configuration/authentication.md (hunk 2 above)
Comment (databyjp) [reply developers/weaviate/current/configuration/authentication.md:128]: Discussed with @dirkkul and agreed to merge as is for now as it is consistent with the existing wording. Let's remove this later if necessary.

**Pull request merged**